packages/twenty-front/src/modules/favorites/components/CurrentWorkspaceMemberFavoritesFolder.tsx

```diff
@@ -0,0 +1,206 @@
+import { useIsPrefetchLoading } from '@/prefetch/hooks/useIsPrefetchLoading';
+import { NavigationDrawerAnimatedCollapseWrapper } from '@/ui/navigation/navigation-drawer/components/NavigationDrawerAnimatedCollapseWrapper';
+import { NavigationDrawerInput } from '@/ui/navigation/navigation-drawer/components/NavigationDrawerInput';
+import { NavigationDrawerItem } from '@/ui/navigation/navigation-drawer/components/NavigationDrawerItem';
+import { NavigationDrawerSection } from '@/ui/navigation/navigation-drawer/components/NavigationDrawerSection';
+import { NavigationDrawerSectionTitle } from '@/ui/navigation/navigation-drawer/components/NavigationDrawerSectionTitle';
+import { useNavigationSection } from '@/ui/navigation/navigation-drawer/hooks/useNavigationSection';
+import { useIsMobile } from '@/ui/utilities/responsive/hooks/useIsMobile';
+import { useTheme } from '@emotion/react';
+import styled from '@emotion/styled';
+import { useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useRecoilState, useRecoilValue } from 'recoil';
+import {
+  Avatar,
+  IconFolder,
+  IconFolderPlus,
+  IconHeartOff,
+  isDefined,
+} from 'twenty-ui';
+
+const StyledContainer = styled(NavigationDrawerSection)`
+  width: 100%;
+`;
+
+const StyledAvatar = styled(Avatar)`
+  :hover {
+    cursor: grab;
+  }
+`;
+
+const StyledNavigationDrawerItem = styled(NavigationDrawerItem)`
+  :active {
+    cursor: grabbing;
+
+    .fav-avatar:hover {
+      cursor: grabbing;
+    }
+  }
+`;
+
+const StyledIconHeartOff = styled(IconHeartOff)<{ isMobile: boolean }>`
+  visibility: ${({ isMobile }) => (isMobile ? 'visible' : 'hidden')};
+  color: ${({ theme }) => theme.color.red};
+
+  .navigation-drawer-item:hover & {
+    visibility: visible;
+  }
+`;
+
+export const CurrentWorkspaceMemberFavoritesFolders = () => {
+  const [activeFolderId, setActiveFolderId] = useState<string | null>(null);
+  const { createFolder, favoriteFolder } = useFavoriteFolders();
+  const { deleteFavorite } = useFavorites();
+  const isMobile = useIsMobile();
+  const [folderName, setFolderName] = useState('');
+  const currentPath = useLocation().pathname;
+  const [isFavoriteFolderCreating, setIsFavoriteFolderCreating] =
+    useRecoilState(isFavoriteFolderCreatingState);
+  const theme = useTheme();
+  const currentWorkspaceMember = useRecoilValue(currentWorkspaceMemberState);
+  const { favorites, handleReorderFavorite, favoritesByFolder } =
+    useFavorites();
+  const loading = useIsPrefetchLoading();
+  const { toggleNavigationSection, isNavigationSectionOpenState } =
+    useNavigationSection('Favorites');
+  const isNavigationSectionOpen = useRecoilValue(isNavigationSectionOpenState);
+
+  const handleFolderNameChange = (value: string) => {
+    setFolderName(value);
+  };
+
+  const handleSubmitFolder = async (value: string) => {
+    const trimmedValue = value.trim();
```

Comment:
I didn't dig enough into the code to understand it but I don't understand why you have code that looks similar twice (with trimming, renaming, etc)

Reply:
this also needs a refactor, thanks for pointing it out